SourceGitlab/SourceGitlab.php

```diff
@@ -147,38 +147,35 @@ public function update_repo( $p_repo ) {
 		$f_hub_root = gpc_get_string( 'hub_root' );
 		$f_hub_repoid = gpc_get_string( 'hub_repoid' );
 		$f_hub_reponame = gpc_get_string( 'hub_reponame' );
+		$f_hub_app_secret = gpc_get_string( 'hub_app_secret' );
+
+		# Update info required for getting the repoid
+		$p_repo->info['hub_root'] = $f_hub_root;
+		$p_repo->info['hub_reponame'] = $f_hub_reponame;
+		$p_repo->info['hub_app_secret'] = $f_hub_app_secret;
 
-		# Getting the repoid doesnt seem to work with the latest gitlab - but we can get all the
-		# repos the current user can access and go through them to find the correct id
-		if( empty( $f_hub_repoid ) ) {
+		# Getting the repoid from reponame
+		if( !is_numeric( $f_hub_repoid ) && !empty( $f_hub_reponame ) ) {
 			$t_hub_reponame_enc = urlencode( $f_hub_reponame );
-			$t_uri = $this->api_uri( $p_repo, "projects" );
+			$t_uri = $this->api_uri( $p_repo, "projects/$t_hub_reponame_enc" );
 			$t_member = null;
 			$t_json = json_url( $t_uri, $t_member );
 
-			$f_hub_repoid = 'RepoName is invalid';
```

Comment:
Hi I am getting this error on adding gitlab repo. No help available online.
Please help. I will delete this comment after one sees this. I am not giving anything in API key

http://192.168.0.205/umang/sample.git
array (size=5)
  'hub_root' => string 'http://192.168.0.205' (length=20)
  'hub_repoid' => string 'RepoName is invalid' (length=19)
  'hub_reponame' => string 'umang/sample' (length=12)
  'hub_app_secret' => string '' (length=0)
  'master_branch' => string 'master' (length=6)